### DISCUSSION TITLE
How to set up dataset and model for TMDD model for total drug and total target

### DISCUSSION THREAD

**Discussion by Ravua1992** [nlmixr2/nlmixr2#277] [Q&A]
Hello, 

I am trying to fit total drug and total soluble target data for a MAB. I have an IV infusion for 1 hour. 
I am not using a CMT column and trying to use only DVID to fit the data. This is because,  I am writing the free drug and free target equation and I want to fit the total drug and total target.  

I am setting up the dataset in following way: 

DVID= "cmab_cc2" and DVID= "solt"

```
    d/dt(rxddta1) <-  -k10 * rxddta1 - k12 * rxddta1 + k21 * rxddta2 - kon*rxddta1*rxddta3 + koff*rxddta4*vc
    d/dt(rxddta2) <- k12 * rxddta1 - k21 * rxddta2
    d/dt(rxddta3) <- KSYN - KDEG*rxddta3 - kon*rxddta3*rxddta1/vc + koff*rxddta4
    d/dt(rxddta4) <- kon*rxddta3*rxddta1/vc - koff*rxddta4 - KINT*rxddta4

 mab_fc2 = rxddta1/scale1 +rxddta4
    solt= rxddta3+rxddta4

cmab_cc2 = (mab_fc2) * MWKDA
    sol_secam5 = (solcea)* MWKDA_ST
    mab_cc2 ~ prop(prop.err1)  | cmab_cc2
    solt ~ prop(prop.err2) | solt

```

When I print the model out I see that the DVs are being considered CMT 5 and CMT 6. Any suggestions how to set it up correctly ? 

```

States ($state or $stateDf): 
  Compartment Number Compartment Name
1                  1          rxddta1
2                  2          rxddta2
3                  3          rxddta3
4                  4          rxddta4
 ── Multiple Endpoint Model ($multipleEndpoint): ──  
        variable                       cmt                       dvid*
1   cmab_cc2 ~ …   cmt='cmab_cc2' or cmt=5   dvid='cmab_cc2' or dvid=1
2 solt ~ …               cmt='solt' or cmt=6             dvid='solt' or dvid=2
  * If dvids are outside this range, all dvids are re-numered sequentially, ie 1,7, 10 becomes 1,2,3 etc

```

also would it be possible to implement B2 method for the observed baseline value of soluble target ? showing nonmem code below as example. 

```
  RTOT1  = RTOT1_OBS*(1.0 + ETA(15)*CCVRES) + ETA(16)*ADDRES
```

Thank you

**Comment by mattfidler** [comment]
> When I print the model out I see that the DVs are being considered CMT 5 and CMT 6. Any suggestions how to set it up correctly ?

This is setup correctly for the `DVID`s you mention.  Note that `nlmixr2`/`rxode2` does a dataset translation and will add the `CMT` as described in the printout.  The `DVID` and `CMT` are two different ways to say the same thing.

**Reply by mattfidler** [reply]
As far as the `b1` method, it seems that you are trying to have individual differences of the residuals based on the baseline? (I haven't used the `b1` method personally yet).  

If so, this *could* work with `focei` only by specifying something like:
```r
  a1 <- addres*abs(eta.addres)
  b1 <- cvvres*abs(eta.cvvres)
  rtot1 ~ add(a1) + prop(b1)
```

Note that you would have to verify that you are modeling the standard deviations (which would mean they have to be positive).

This is a feature that is not well tested (and may not work as well as it could in future revisions).

**Reply by Ravua1992** [reply]
Thank you so much for confirming! 
I will try this way. 
```

  a1 <- addres*abs(eta.addres)
  b1 <- cvvres*abs(eta.cvvres)
  rtot1 ~ add(a1) + prop(b1)

```

I am trying to reproduce a model from nonmem where IMP estimation method was used. In this B2 method was used for soluble target baseline measurements due to some issues with the assay. 

Would this mean that I would need to adjust the DVID as well now ? Since Rtot1 will also be estimated. The remaining two dvid's  will be dvid== 2 and dvid== 3 now. orginally dvid==1 and dvid==2.

Apologies for another question, is there any equivalent of the IMP estimation method that i can use? I am currently using focei method.

**Reply by mattfidler** [reply]
> I am trying to reproduce a model from nonmem where IMP estimation method was used. In this B2 method was used for soluble target baseline measurements due to some issues with the assay.

It is not clear to me that you are attempting to model the `rtot1` in the data you are specifying.  Note that you would have to have data to support this `rtot1` for it to be modeled.

> Would this mean that I would need to adjust the DVID as well now ? Since Rtot1 will also be estimated. The remaining two dvid's will be dvid== 2 and dvid== 3 now. orginally dvid==1 and dvid==2.

In theory if you are using named `dvid`s you should not need to rename or change anything.

> I am trying to reproduce a model from nonmem where IMP estimation method was used. 

You may want to try `nonmem2rx` which is accessible from `babelmixr2`

```r
library(babelmixr2)
nonmem2rx("path/to/your/nonem.lst")
# You can also specify the parameter names instead of letting `nonmem2rx` decide for you.
```

>  is there any equivalent of the IMP estimation method that i can use?

No, not currently.  We have `focei`, `saem` and `nlme` for built-in optimization methods.  Babelmixr2 can run NONMEM from the nlmixr2 model/data and use NONMEM's `IMP`, though I don't think that is what you meant here. (You can also run Monolix from a `nlmixr2` model)
